drivers/sensor/bme680/bme680.h

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2016, 2017 Intel Corporation
+ * Copyright (c) 2017 IpTronix S.r.l.
+ * Copyright (c) 2018 Bosch Sensortec GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SENSOR_BME680_H__
+#define __SENSOR_BME680_H__
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#define BME680_CHIP_ID                  0x61
+
+#define BME680_LEN_FIELD                15
+#define BME680_LEN_COEFF_ALL            42
+#define BME680_LEN_COEFF1               23
+#define BME680_LEN_COEFF2               14
+#define BME680_LEN_COEFF3               5
+
+#define BME680_REG_COEFF3               0x00
+#define BME680_REG_FIELD0               0x1d
+#define BME680_REG_IDAC_HEAT0           0x50
+#define BME680_REG_RES_HEAT0            0x5A
+#define BME680_REG_GAS_WAIT0            0x64
+#define BME680_REG_SHD_HEATR_DUR        0x6E
+#define BME680_REG_CTRL_GAS_0           0x70
+#define BME680_REG_CTRL_GAS_1           0x71
+#define BME680_REG_CTRL_HUM             0x72
+#define BME680_REG_CTRL_MEAS            0x74
+#define BME680_REG_CONFIG               0x75
+#define BME680_REG_MEM_PAGE             0xf3
+#define BME680_REG_UNIQUE_ID            0x83
+#define BME680_REG_COEFF1               0x8a
+#define BME680_REG_CHIP_ID              0xd0
```

Comment:
In order to have leaner code, like in the BME680 API, addresses above `0x7f` are on a different page when operation over the SPI interface. Hence, instead of checking in each API for the interface selected or even having separate drivers for SPI and I2C, denoting the pages in this manner allows for a unified driver. This taken care of in the `bme680_get_regs` and `bme680_set_regs` APIs which allows the rest of the code to only handle the I2C addresses.